iroh-net/src/util.rs

```diff
@@ -10,7 +10,7 @@ use futures::FutureExt;
 
 /// A join handle that owns the task it is running, and aborts it when dropped.
 #[derive(Debug, derive_more::Deref)]
-pub(crate) struct AbortingJoinHandle<T>(tokio::task::JoinHandle<T>);
+pub struct AbortingJoinHandle<T>(pub tokio::task::JoinHandle<T>);
```

Comment:
why make the field pub?

Reply:
Because I now use this from iroh to make sure the optional gc task is aborted on drop. This would be another one for the util crate.